holoviews/plotting/bokeh/util.py

```diff
@@ -226,15 +242,14 @@ def delete_refs(obj, locs, delete):
             return None
         new_obj = {}
         for k, v in list(obj.items()):
-            if k in locs:
-                ref = delete_refs(v, locs, delete)
-                if ref is not None:
-                    new_obj[k] = ref
-            else:
-              new_obj[k] = v
+            if k in ['data', 'palette'] or (k == 'attributes' and not get_ids(v)):
```

Comment:
Agreed, I don't envision them changing but breaking them out probably makes sense regardless. As a quick explanation, there are two bits of data that are sent to update a plot:

1) The events: These contain the actual attributes that have changed and need to be updated.
2) The references: These contain references to the objects in the plot but also duplicate the data.

This PR filters filters the various data attributes in the references leaving only the id references that are actually needed. This means that the ``data``, the ``palette`` and all attributes that do not contain references to other objects are deleted. Passing in the list of attributes that should be dropped explicitly makes sense so I'll do that now.